docs(link): add mention about how App Router prefetches on hover (#70679)
vercel/next.js: docs/02-app/02-api-reference/01-components/link.mdx

<AppOnly>

> Prefetching happens when a `<Link />` component enters the user's viewport (initially or through scroll). Next.js prefetches and loads the linked route (denoted by the `href`) and its data in the background to improve the performance of client-side navigation's. If the prefetched data has expired by the time the `<Link />` is clicked, Next.js will attempt to prefetch again on hover. **Prefetching is only enabled in production**.

Reviewer: ```suggestion
Prefetching happens when a `<Link />` component enters the user's viewport (initially or through scroll). Next.js prefetches and loads the linked route (denoted by the `href`) and its data in the background to improve the performance of client-side navigation's. If the prefetched data has expired by the time the user hovers over the link, Next.js will attempt to prefetch it again. **Prefetching is only enabled in production**.
```
